seq/seq.c

```diff
@@ -40,26 +70,214 @@ int main(int argc, char *argv[]){
 		usage();
 	}
 
-	// If there's no separator set, defaults to a line break (\n)
-	separator = ( strncmp(separator, "\0", sizeof("\0")) == 0 ) ? "\n" : separator;
-
-	// If argc equals 1, stop will be defined as the first command line argument
-	// and start and step will be defaulted to 1.
-	// If argc is 2, stop will be defined as the second command line argument and
-	// start as the first.
-	// If argc is 3, stop will be defined as the third, start as the first and
-	// step as the second.
-	stop = ( argc == 1 ) ? atoi(argv[0]) : ( argc == 3 ) ? atoi(argv[2]) : atoi(argv[1]);
-	start = ( argc == 1 ) ? 1 : atoi(argv[0]);
-	step = ( argc < 3 ) ? 1 : atoi(argv[1]);
-
-	for ( count = start; count <= stop; count += step ) {
-		separator = (count == stop) ? "\n" : separator;
-		printf("%d%s", count, separator);
+	/* 
+	 * If argc equals 1, stop will be defined as the first command line argument
+	 * and start and step will be defaulted to 1.
+	 * If argc is 2, stop will be defined as the second command line argument and
+	 * start as the first.
+	 * If argc is 3, stop will be defined as the third, start as the first and
+	 * step as the second.
+	 */
+	stop = ( argc == 1 )
+		? atof(argv[0])
+		: ( argc == 3 )
+		? atof(argv[2])
+		: atof(argv[1]);
+	start = ( argc == 1 )
+		? 1
+		: atof(argv[0]);
+	step = ( argc < 3 )
+		? 1
+		: atof(argv[1]);
+	
+	if (step == 0) {
+		pfmt(stderr, MM_ERROR,
+			"%s: increment can not be zero.\n",
+			progname);
+		exit(1);
+	}
+
+	format = buildfmt();
+
+	/* If there's no separator set, defaults to a line break (\n). */
+	separator = (strncmp(separator, "", sizeof(char *)) == 0)
+			? "\n" 
+			: separator;
+
+	for (count = start; count <= stop; count += step) {
+		/* 
+		 * If the count has come to the end or if the next sum is
+		 * larger than stop, default separator back to '\n'.
+		 */
+		separator = (count == stop || (count + step) > stop)
+				? "\n" : separator;
+		printf(format, count, separator);
+	}
+	free(format);
+
+	exit(0);
```

Comment:
Just a simple formality.

Reply:
The `int` approach is clearly documented on all posixes, it's not insightful to disguise personal taste (or lack of) on the unexplained concept of "formality"

Reply:
I just wanted to [joke around with the name of a film](https://en.wikipedia.org/wiki/A_Pure_Formality).

In any case, this is just a "loan" that I ended up doing from Google's Go programming style and Plan 9, not a big deal. I do not think this could cause any problem and I think it looks better to one who's studying the code.